src/strands/ir_builders.js

```diff
@@ -463,40 +464,72 @@ export function swizzleTrap(id, dimension, strandsContext) {
           }
         }
     },
-    set(target, property, value, receiver) {
-      for (const set of swizzleSets) {
-        const chars = [...property];
-        const valid =
-          chars.every(c => set.includes(c)) 
-          && new Set(chars).size === chars.length
-          && target.dimension >= chars.length;
-        if (valid) {
-          const originalNode = DAG.getNodeDataFromID(strandsContext.dag, target.id);
-          const changed = Object.fromEntries(
-            chars.map((c, i) => [set.indexOf(c), i])
-          )
-          if (typeof value === 'number') {
-            value = new Array(chars.length).fill(value);
-          } 
-          const newValues = mapPrimitiveDepsToIDs(
-            strandsContext,
-            {baseType: originalNode.baseType, dimension: null}, 
-            value
-          ).mappedDependencies;
-          const newDeps = new Array(originalNode.dimension);
-          for (let i = 0; i < target.dimension; i++) {
-            console.log(changed[i])
-            if (changed[i] !== undefined) {
-              newDeps[i] = newValues[changed[i]];
-            }
+  set(target, property, value, receiver) {
+    for (const basis of swizzleSets) {
+      const chars = [...property];
+      const valid =
+        chars.every(c => basis.includes(c)) &&
+        new Set(chars).size === chars.length &&
+        target.dimension >= chars.length;
+
+      if (!valid) continue;
+
+      const dim = target.dimension;
+
+      const lanes = new Array(dim);
```

Comment:
are "lanes" like the underlying values for each dimension, which may go under different name aliases? e.g. lane 0 could be `x` or `r`? (If so maybe add a comment explaining)